ballerina/json_schema.bal

```diff
@@ -0,0 +1,129 @@
+// Copyright (c) 2025, WSO2 LLC. (https://www.wso2.com).
+//
+// WSO2 LLC. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied. See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+# Defines constraints for JSON schema string type.
+#
+# + maxLength - The maximum number of characters allowed in the string.
+# + minLength - The minimum number of characters required in the string.
+# + pattern - A regular expression that the string must match.
+# + format - The format that the string must adhere to (e.g., email, date-time).
+public type StringValidationRec record {|
```

Comment:
```suggestion
public type StringValidations record {|
```

Is this really validation, though? Can't we use `StringConstraints`, similar to the constraint module?